python/ffsim/protocols/trace_protocol.py

```diff
@@ -45,3 +51,69 @@ def trace(obj: Any, norb: int, nelec: tuple[int, int]) -> float:
         "The object did not have a _trace_ method that returned the trace "
         "or a _diag_ method that returned its diagonal entries."
     )
+
+
+def _trace_term(
+    op: tuple[tuple[bool, bool, int], ...], norb: int, nelec: tuple[int, int]
+):
```

Comment:
Add a return type annotation. Now that I think of it, I suppose the return type of the `trace` function should probably be `complex`.